CONTRIBUTING.rst

```diff
@@ -0,0 +1,500 @@
+
+.. _Getting Started Guide:
+   https://www.zephyrproject.org/doc/getting_started/getting_started.html
+
+As a Linux developer, you should also be familiar with common development
+tools such as Git, GitHub, and make. The preferred development environment is
+Linux.
+
+If you haven’t already done so, you’ll need to create a (free) GitHub account
+on http://github.com and have Git tools available on your development system.
+
+Repository layout
+*****************
+
+To clone the main Zephyr Project repository use::
+
+    git clone https://github.com/zephyrproject-rtos/zephyr
+
+(You may find `using ssh access to GitHub`_ is more convenient since you
+can connect to GitHub without supplying your username or password at each
+visit.)
+
+.. _using SSH access to GitHub:
+   https://help.github.com/articles/connecting-to-github-with-ssh/
+
+The Zephyr project directory structure is describe in `Source Tree Structure`_
+documentation. In addition to the Zephyr kernel itself, you'll also find the
+sources for technical documentation, sample code, supported board
+configurations, and a collection of subsystem tests.  All of these are
+available for developers to contribute to and enhance.
+
+.. _Source Tree Structure:
+   https://www.zephyrproject.org/doc/kernel/overview/source_tree.html
+
+Logging Bugs and Enhancement Requests
+*************************************
+
+.. _Zephyr Project Issues: https://jira.zephyrproject.org
+
+Before starting on a patch, first check in our Jira `Zephyr Project Issues`_
+system to see what’s been reported on the issue you’d like to address.  Have a
+conversation on the mailing list (or IRC) to see what others think of your
+issue (and proposed solution).  You may find others that have encountered the
+issue you're finding, or with similar ideas for changes or additions.  Send a
+message to the mailing list to introduce and discuss your idea with the
+development community.
+
+It's always a good practice to search for existing or related issues before
+submitting your own. When you submit an issue (bug or feature request), the
+triage team will review and comment on the submission, typically within a few
+business days.
+
+Development Tools and Git Setup
+*******************************
+
+Signed-off-by
+=============
+
+The name in the commit message ``Signed-off-by:`` line and your email must
+match the change authorship information. Make sure your *.git/config* is set
+up correctly::
+
+   $ git config --global user.name "David Developer"
+   $ git config --global user.email "david.developer@company.com"
+
+gitlint
+=========
+
+When you submit a pull-request to the project, a series of checks are
+performed to verify your commit messages meet the requirements, the same step
+done during the CI process can be performed locally using the the *gitlint*
+command.
+
+Install gitlint and run it locally in your tree and branch where your patches
+have been committed::
+
+     $ sudo pip install gitlint
+     $ gitlint
```

Comment:
This only checks HEAD (the most recent commit). You might want to adjust the text to recommend running it after each commit, or using the `--commits` option to specify a commit range covering all the development branches to be submitted.